### DISCUSSION TITLE
iRODS and ubuntu 18, 20, 22

### DISCUSSION THREAD

**Discussion by Danny-dK** [UtrechtUniversity/yoda#410] [Q&A] [answered]
I see in October on https://irods.org/2023/10/irods-development-update-october-2023/ that there is support for Ubuntu 22.04 using iRODS 4.3. On https://www.uu.nl/en/research/yoda/guide-to-yoda/i-am-using-yoda/using-icommands-for-large-datasets it is indicated that Yoda currently uses 4.2.12 and that "iRODS version 4.2.12 currently supports Ubuntu 18.04 LTS." 
I have pretty much used 20.04LTS only and worked without fail (but have not done very difficult things). I'm assuming the official statement is that Yoda is mainly supported with Ubuntu 18.04, but unofficially are there any things to take into account when using Ubuntu 22.04 and the latest icommands version?

**Comment by stsnel** [comment, marked as answer]
Hi Danny,

RENCI does not support mismatched iCommands / iRODS server combinations, and such combinations are not tested as a part of the iRODS or Yoda QA processes. If a combination of different versions does not work as expected, we generally have no or few practical options to fix this.  Although I am not aware of any specific issues with using iCommands 4.3.1 in combination with iRODS 4.2.12, we've had some experiences in the past where another non-matching iCommands/server combination appeared to work at first, but it turned out to result in problems that were not immediately obvious.

Alternatives you could consider:
- The GoCommands (https://github.com/cyverse/gocommands) are easier to get working on different distribution versions. We also generally have more troubleshooting options here, since (among other things) the dependency situation is quite simple relative to iCommands and the release cycle is much faster.
- Installing iCommands 4.2.12 on Ubuntu 20.04 LTS could be considered. Installing it manually is admittedly a bit of a hairy process, but [the Vagrant-irods iCommands VM setup](https://github.com/UtrechtUniversity/vagrant-irods) also supports Ubuntu 20.04 LTS

Best,
Sietse
